cluster/swarm/cluster.go

```diff
@@ -220,8 +220,12 @@ func (c *Cluster) createContainer(config *cluster.ContainerConfig, name string,
 
 	c.scheduler.Unlock()
 
-	log.WithFields(log.Fields{"NodeName": n.Name, "NodeID": n.ID}).Debugf("Scheduling container %s to ", name)
 	container, err := engine.CreateContainer(config, name, true, authConfig)
+	containerFlag := name
+	if containerFlag == "" && err == nil {
+		containerFlag = stringid.TruncateID(container.ID)
+	}
+	log.WithFields(log.Fields{"NodeName": n.Name, "NodeID": n.ID}).Debugf("Scheduling container %s to ", containerFlag)
```

Comment:
@xianlubird 
I think that, when the err is not nil , we should handle this kind of situation. In the code above, no matter what err is, we will log the scheduling details.